mongodbatlas/cloud_provider_access.go

```diff
@@ -133,13 +138,13 @@ func (s *CloudProviderAccessServiceOp) AuthorizeRole(ctx context.Context, groupI
 // DeauthorizeRole deauthorizes an AWS Assumed IAM role.
 //
 // See more: https://docs.atlas.mongodb.com/reference/api/cloud-provider-access-deauthorize-one-role/
-func (s *CloudProviderAccessServiceOp) DeauthorizeRole(ctx context.Context, groupID, roleID string) (*Response, error) {
+func (s *CloudProviderAccessServiceOp) DeauthorizeRole(ctx context.Context, groupID, roleID string, request *CloudProviderDeauthorizationRequest) (*Response, error) {
```

Comment:
Sorry as this is totally in me but could we make the third param just a `string` as there's no need to serialize it, something like
```suggestion
func (s *CloudProviderAccessServiceOp) DeauthorizeRole(ctx context.Context, groupID, providerName, roleID string) (*Response, error) {
```
if you prefer to keep the struct `CloudProviderDeauthorizationRequest`, we could move all there, groupID, and roleID, and no need to use json tags as this won't be serialized